cloudify_aws/ec2/resources/transit_gateway.py

```diff
@@ -0,0 +1,308 @@
+    '''
+
+    def __init__(self, ctx_node, resource_id=None, client=None, logger=None):
+        EC2Base.__init__(self, ctx_node, resource_id, client, logger)
+        self.type_name = RESOURCE_TYPE
+
+    @property
+    def properties(self):
+        '''Gets the properties of an external resource'''
+        params = {TG_ATTACHMENT_IDS: [self.resource_id]}
+        try:
+            resources = \
+                self.client.describe_transit_gateway_vpc_attachments(**params)
+        except ClientError:
+            pass
+        else:
+            return None if not resources else resources.get(
+                TG_ATTACHMENTS, [None])[0]
+        return None
+
+    @property
+    def status(self):
+        '''Gets the status of an external resource'''
+        props = self.properties
+        if not props:
+            return None
+        return props['State']
+
+    def create(self, params):
+        '''
+            Create a new AWS EC2 Transit Gateway Attachment.
+        '''
+        return self.make_client_call(
+            'create_transit_gateway_vpc_attachment', params)
+
+    def accept(self, params):
+        '''
+            Create a new AWS EC2 Transit Gateway Attachment.
+        '''
+        return self.make_client_call(
+            'accept_transit_gateway_vpc_attachment', params)
+
+    def delete(self, params=None):
+        '''
+            Deletes an existing AWS EC2 Transit Gateway Attachment.
+        '''
+        return self.make_client_call(
+            'delete_transit_gateway_vpc_attachment', params)
+
+
+@decorators.aws_resource(EC2TransitGateway, resource_type=RESOURCE_TYPE)
+def prepare(ctx, iface, resource_config, **_):
+    '''Prepares an AWS EC2 Transit Gateway'''
+    # Save the parameters
+    ctx.instance.runtime_properties['resource_config'] = resource_config
+
+
+@decorators.aws_resource(EC2TransitGateway, RESOURCE_TYPE)
+@decorators.wait_for_status(status_good=['available'],
+                            status_pending=['pending'])
+@decorators.tag_resources
+def create(ctx, iface, resource_config, **_):
+    '''Creates an AWS EC2 Transit Gateway'''
+    params = utils.clean_params(
+        dict() if not resource_config else resource_config.copy())
+
+    # Actually create the resource
+    create_response = iface.create(params)[TG]
+    ctx.instance.runtime_properties['create_response'] = \
+        utils.JsonCleanuper(create_response).to_dict()
+
+    transit_gateway_id = create_response.get(TG_ID, '')
+    iface.update_resource_id(transit_gateway_id)
+    utils.update_resource_id(ctx.instance, transit_gateway_id)
+
+
+@decorators.aws_resource(EC2TransitGateway, RESOURCE_TYPE,
+                         ignore_properties=True)
+@decorators.untag_resources
+def delete(iface, resource_config, **_):
+    '''Deletes an AWS EC2 Transit Gateway'''
+
+    params = dict() if not resource_config else resource_config.copy()
+
+    if TG_ID not in params:
+        params.update({TG_ID: iface.resource_id})
```

Comment:
Why you send all the params? As I see in line 815 in plugin.yaml the params will be for the create operation.

Reply:
Because we need them.